lib/data/multipleBackendGateway.js

```diff
@@ -73,35 +36,9 @@ const multipleBackendGateway = {
         }
         // client is AWS SDK
         if (client.clientType === 'aws_s3') {
-            const awsKey = _createAwsKey(keyContext.bucketName,
-               keyContext.objectKey, client.bucketMatch);
-            // TODO: if object to be encrypted, use encryption
-            // on AWS
-            return client.upload({
-                Bucket: client.awsBucketName,
-                Key: awsKey,
-                Body: stream,
-                Metadata: keyContext.metaHeaders,
-                ContentLength: size,
-            },
-               (err, data) => {
-                   if (err) {
-                       const log = createLogger(reqUids);
-                       log.error('err from data backend',
-                       { error: err, dataStoreName: client.dataStoreName });
-                       return callback(errors.InternalError);
-                   }
-                   const dataRetrievalInfo = {
-                       key: awsKey,
-                       dataStoreName: client.dataStoreName,
-                       dataStoreType: client.clientType,
-                       // because of encryption the ETag here could be
-                       // different from our metadata so let's store it
-                       // TODO: let AWS handle encryption
-                       dataStoreETag: data.ETag,
-                   };
-                   return callback(null, dataRetrievalInfo);
-               });
+            const log = createLogger(reqUids);
+            return awsClient.put(client, stream, size, keyContext, log,
```

Comment:
The changes are improving clarity already, but can't we go further and get rid of the `client.clientType === 'aws_s3'` checks in this file?

Unless I miss something, this should be doable by making the `awsClient` compatible with the other backend interfaces (e.g. we may call `createLogger` in the backend since it does not take a `log` param).